common/optimiser/Optimiser.java

```diff
@@ -46,6 +46,7 @@ public class Optimiser {
     private MPSolverParameters parameters;
     private Status status;
     private boolean hasSolver;
+    private double objectiveValueCache;
```

Comment:
I think we should refactor this to be `objectiveValue`, and always record the solver's objective value into it every time. It looks a bit odd to have a variable called `cache` (caches tend to be things we can remove and still maintain correctness) - but if it is always used that would make sense.